### ISSUE TITLE
Feature Request: Edit the description of the Annotations

### ISSUE THREAD

**Issue by GrazianoCapelli** [BasicAirData/GPSLogger#158]
A user may need to change the label of the annotations.
To date there is no way to do it: the user must export the track and then edit the GPX files by hand.
A function to edit the annotations would be useful, either to correct typing error on the fly or in case of change of mind.

I think that, prior to / during the implementation of this feature, we should evaluate if the editing of annotations is feasible also without have a map.

_Time for implementation ideas, mock-ups, comments..._

**Comment by TVEgit** [comment]
I believe modifying an annotation should be doable. It is independant from map viewing.

**Comment by icrf2000** [comment]
I started using the app yesterday and immediately found it would be useful to be able to edit/delete waypoints/annotations. So from me, thumbs up for your idea. A list of existing WPT's could open when tapping the "annotations" button on the TRACK tab which currently only shows the number of annotations.

**Comment by GrazianoCapelli** [comment]
I agree with you, the annotations tile is the best place to access to annotations of the current track, I thought to add a small edit icon on that tile, like the _copy_ icon on coordinates.
But we should find a good place to access the list of annotations also from the tracklist, in order to allow the user to edit the annotations of all tracks, not only the active one. I think that the best place could be the _edit detail_ dialog, but I haven't found yet an elegant position for it. We should start with a mock-up image before to start the implementation.
Of course ideas and suggestions are really appreciated.

Anyway, as first implementation we could start to add the edit feature only for the active track, and as a second step we could expand the feature to all the tracks of the tracklist.

**Comment by TVEgit** [comment]
I think the most elegant way to implement is to find a way to load an archived track into the track panel, just like it is before the stop button is pressed.
By doing that waypoint edition would be done exactly the same way.

**Comment by icrf2000** [comment]
I created a mockup of where the feature could be exposed in the tracklist tab. Sorry, I am not a good Gimp painter...

I'm not yet sure what's the technical difference between the currently running track (shown in the "TRACK" tab AND the "TRACKLIST" tab) and the old, closed ones (shown only in the "TRACKLIST" tab). I exported the database from my device and gave it a glance to compare. I see no difference, so actually it should be possible to treat old tracks and current track in the same way.

![edit_annotations_tracklist_mockup](https://github.com/BasicAirData/GPSLogger/assets/11777822/b7232fb1-5ac8-4743-a20b-996f8119b784)

**Comment by GrazianoCapelli** [comment]
As first implementation we can start to add the possibility to edit the placemarks description of the current active track.
We can add an edit icon to the annotations tile (like we already do into the Coordinates one) as shown into the image here below:

![EditAnnot](https://github.com/user-attachments/assets/ca502e58-740e-4eca-9ad8-59346ccffa27)

the icon opens the list of the annotations of the current track in a dialog or a new activity, showing a progressive number and the description of each one.
Clicking an annotation allows to change the description text.

Maybe a good idea would be to start with a mockup.
